diff --git a/src/libtracker-common/tracker-hal.c b/src/libtracker-common/tracker-hal.c
index b539584..6628683 100644
--- a/src/libtracker-common/tracker-hal.c
+++ b/src/libtracker-common/tracker-hal.c
@@ -604,6 +604,13 @@ hal_device_is_removable (TrackerHal  *hal,
 	LibHalDrive	*drive;
 	gboolean	 removable;
 
+	/* FIXME: This is a hack for Maemo only. For some reason this
+	 * function says the MMC is NOT removable, for now we always
+	 * return TRUE because there is only one HAL device and that
+	 * is the MMC.
+	 */     
+	return TRUE;
+
 	if (!device_file) {
 		return FALSE;
 	}
