diff --git a/src/libtracker-common/tracker-hal.c b/src/libtracker-common/tracker-hal.c
index b6942d3..dc5f763 100644
--- a/src/libtracker-common/tracker-hal.c
+++ b/src/libtracker-common/tracker-hal.c
@@ -732,6 +732,13 @@ hal_device_is_removable (TrackerHal  *hal,
 	LibHalDrive	*drive;
 	gboolean	 removable;
 
+	/* FIXME: This is a hack for Maemo only. For some reason this
+	 * function says the MMC is NOT removable, for now we always
+	 * return TRUE because there is only one HAL device and that
+	 * is the MMC.
+	 */     
+	return TRUE;
+
 	if (!device_file) {
 		return FALSE;
 	}
@@ -903,6 +910,13 @@ hal_device_add (TrackerHal   *hal,
 		return TRUE;
 	}
 
+	/* Ignore /home partition for the maemo platform */
+	if (mount_point &&
+	    (strcmp (mount_point, "/home") == 0 ||
+	     strcmp (mount_point, "/home/") == 0)) {
+		return TRUE;
+	}
+
 	/* Make sure we watch changes to the mount/umount state */
 	libhal_device_add_property_watch (priv->context, udi, &error);
 
