diff --git a/data/modules/files.module b/data/modules/files.module
index 4b26c83..6fac7ee 100644
--- a/data/modules/files.module
+++ b/data/modules/files.module
@@ -8,7 +8,7 @@ RecurseDirectories=
 
 [Ignored]
 Directories=po;CVS;.svn;.git;core-dumps;
-Files=*~;*.o;*.la;*.lo;*.loT;*.in;*.csproj;*.m4;*.rej;*.gmo;*.orig;*.pc;*.omf;*.aux;*.tmp;*.po;*.vmdk;*.vm*;*.nvram;*.part;*.rcore.lzo;autom4te;conftest;confstat;Makefile;SCCS;litmain.sh;libtool;config.status;confdefs.h;
+Files=*~;*.o;*.la;*.lo;*.loT;*.in;*.csproj;*.m4;*.rej;*.gmo;*.orig;*.pc;*.omf;*.aux;*.tmp;*.po;*.vmdk;*.vm*;*.nvram;*.part;*.rcore.lzo;autom4te;conftest;confstat;Makefile;SCCS;litmain.sh;libtool;config.status;confdefs.h;*.ra;*.ram;*.rm;*.rmj;*.rmm;*.rmvb;*.rpm;*.rv;*.smil;
 DirectoriesWithContent=backup.metadata;
 
 [Index]
