diff --git a/data/services/default.service b/data/services/default.service
index 942997a..24ff7cd 100644
--- a/data/services/default.service
+++ b/data/services/default.service
@@ -111,8 +111,7 @@ KeyMetadata10=Audio:DateAdded
 KeyMetadata11=Audio:LastPlay
 TabularMetadata=Audio:Title;Audio:Artist;Audio:Album;Audio:Genre;Audio:Duration;Audio:ReleaseDate;
 TileMetadata=Audio:Title;Audio:Artist;Audio:Album;Audio:Genre;Audio:Duration;Audio:ReleaseDate;
-Mimes=application/ogg;
-MimePrefixes=audio/;
+Mimes=audio/mpeg;audio/x-mp3;audio/x-mpeg;audio/mp3;audio/x-wav;audio/wav;audio/x-m4a;audio/x-mp2;audio/mp2;audio/x-amr;audio/AMR;audio/x-ms-wma;audio/basic;audio/mp4;
 ShowServiceFiles=true
 ShowServiceDirectories=true
 HasMetadata=true
@@ -136,8 +135,7 @@ KeyMetadata7=Video:Width
 Keymetadata8=Video:Height
 TabularMetadata=File:Name;Video:Title;Video:Author;Video:Height;Video:Width;Video:Duration;Video:Bitrate;
 TileMetadata=Video:Title;Video:Author;Video:Height;Video:Width;Video:Duration;Video:Bitrate;
-Mimes=application/vnd.rn-realmedia
-MimePrefixes=video/;
+Mimes=video/x-msvideo;video/avi;video/mpeg;video/x-mp4;video/mp4;video/3gpp;video/x-ms-wmv;video/x-ms-asf-plugin;video/quicktime;video/x-ms-asf;
 ShowServiceFiles=true
 ShowServiceDirectories=true
 HasMetadata=true
